src/Documentation/resources/Best_Practices.md

```diff
@@ -0,0 +1,422 @@
+# Best Practices
+
+Orleans was designed in 2014 with the intention of removing low-level
+abstractions from actor model frameworks. Grains (actors stored in memory) are
+used to maintain the lifecycle among other features. Orleans is based on the
+actor model and fits some projects better that others.
+
+## Orleans should be considered when:
+
+-   Significant number (hundreds to millions) of loosely coupled entities
+
+    -   Example: Internet of Things (IoT) – individual devices turning on/off
+
+-   Entities are small enough to be single-threaded
+
+    -   Example: Determine if stock should be purchased based on current price
+
+-   Workload is interactive
+
+    -   Example: request-response, start/monitor/complete
+
+-   More than one server is expected or may be required\\
+
+    -   Orleans runs on a cluster which is expanded by adding servers to expand
+        the cluster
+
+-   Global coordination is not needed or on a smaller scale between a few
+    entities at a time
+
+    -   Orleans maintains its clusters and talks to other clusters. Each cluster
+        is able to communicate through tcp/ip
+
+-   *Various entities are used at different times*
```

Comment:
This is debatable. it is entirely possible to keep all entities of certain type in memory, for example, all users or all devices. And for some applications that's the best. Of course, the number of entities must be reasonable in those case. But in general, there's no requirement for the set of active entities to differ over time, even though that's a popular case.

Reply:
corrected